### DISCUSSION TITLE
response_allowed_while_playing in conjunction with button_html

### DISCUSSION THREAD

**Discussion by kennysmithed** [jspsych/jsPsych#1388] [Q&A] [answered]
Hi,
One of my students pointed out an unexpected behaviour for response_allowed_while_playing (in audio-button-response). This is a great addition to the plugin by the way, very useful.  When the buttons are text it works as expected; when I use button_html to set the buttons up as images, it doesn't - the cursor has the "no entry" sign on it, but the buttons register clicks and allow the trial to be ended with a response while the audio is still playing. Example code for the two button types below - image_button_trial doesn't work as expected, text_button_trial does (excuse the weird stim names!). I had a look at the plugin code but I couldn't figure out the problem. Not sure if the way I am doing image buttons via button_html is problematic, or if it's to do with the way response_allowed_while_playing is implemented interacting with button_html, any advice much appreciated!

```js
var image_button_trial = {type:'audio-button-response',
                          stimulus: 'animal_ear.mp3',
                          button_html: '<button class="jspsych-btn"> <img src="%choice%.jpg" width=250px></button>',
                          choices:['animal_ear','animal_nose'],
                          response_allowed_while_playing:false
                        }


var text_button_trial = {type:'audio-button-response',
                          stimulus: 'animal_ear.mp3',
                          //button_html: '<button class="jspsych-btn"> <img src="%choice%.jpg" width=250px></button>',
                          choices:['animal_ear','animal_nose'],
                          response_allowed_while_playing:false
                        }
```

**Comment by jodeleeuw** [comment, marked as answer]
Hey @kennysmithed ,

We just fixed this yesterday. See #1380 for a solution, and it'll be patched in 6.3.

**Reply by kennysmithed** [reply]
Ah amazing, thanks (and sorry for missing that had already been dealt with).
